### PR TITLE
Update npm scripts, fix routing bugs, upgrade allStops …

### PR DESCRIPTION
**Pull request by zeladada**
…functionality

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,7 +9,8 @@ osrm/
 *.zip
 npm-debug.log
 osrm-backend/*
-*graphhopper*
+*graphhopper/*
+*graphhopper.*
```
Comment (young-k) [inline .gitignore]: what does this catch that wasn't previously caught by `*graphhopper*`?

File: scripts/run.sh [hunk 2]
```diff
@@ -24,9 +24,39 @@
 usage()
 {
     echo "Options:
-    [-p | --prod] use Transit production mode (without running Transit in its own docker image/container)
-    [-s | --setup] do not start Transit when finished
-    [-h | --help] display these options"
+    [-p | --prod]
```
Comment (young-k) [inline scripts/run.sh]: big fan of these changes

File: scripts/run_ghopper.sh [hunk 3]
```diff
@@ -3,9 +3,7 @@
 # grep -E -i ".*(\[main\]|\[INFO\]|erro?r?|exception|at.*(java|graphhopper)|exit*).*" # grep whitelist lines
 # grep -E -i "^((?!download|points?|maven).)*$" # grep blacklist lines
 
-cd docker-compose
 echo "Initializing Graphhopper Services..."
-docker-compose up --no-start --force-recreate ghopper # | grep -v -i 'points?0?\|download'
-docker-compose up | rotatelogs -f -l  -n 5 ./logs/.graphhopper.log 86400 1M
-cd ..
-echo "Exiting"
+docker-compose -f docker-compose/docker-compose.yml up --no-start --force-recreate ghopper # | grep -v -i 'points?0?\|download'
```
Comment (young-k) [inline scripts/run_ghopper.sh]: can we delete the `# grep`?

File: src/utils/AllStopUtils.js [hunk 4]
```diff
@@ -4,8 +4,54 @@ import RequestUtils from './RequestUtils';
 import TokenUtils from './TokenUtils';
 import ErrorUtils from './LogUtils';
 
-let allStops = RequestUtils.fetchRetry(fetchAllStops);
 const HOUR_IN_MS = 1000 * 60 * 60;
+const DEG_EXACT_PRECISION = 6; // 6 degrees of precision is about a 111 mm, is exact point
+const DEG_EQ_PRECISION = 5; // 5 degrees of precision is about a 1.1 meters, is a stop
+const DEG_NEARBY_PRECISION = 4; // 4 degrees of precision is about 11 meters, stop nearby
+const DEG_WALK_PRECISION = 3; // 3 degrees of precision is about 111 meters, stop walkable
+const DEG_KM_PRECISION = 2; // 3 degrees of precision is about 1 km, stop barely walkable
```
Comment (young-k) [inline src/utils/AllStopUtils.js]: big fan of this as well

File: src/server.js [hunk 5]
```diff
@@ -1,13 +1,16 @@
...
 import AlertsUtils from './utils/AlertsUtils';
 import AllStopUtils from './utils/AllStopUtils';
 import API from './Api';
 import RealtimeFeedUtils from './utils/RealtimeFeedUtils';
 import ErrorUtils from './utils/LogUtils';
 import TokenUtils from './utils/TokenUtils';
+import GhopperUtils from './utils/GraphhopperUtils';
+
+// eslint-disable-next-line no-console
+console.log('\x1b[36m%s\x1b[0m', 'Initializing data and waiting for Graphhopper services...');
```
Comment (young-k) [inline src/server.js]: wtf is this console log

File: src/test/Unit.test.js [hunk 6]
```diff
@@ -20,7 +25,7 @@ describe('Route unit tests', () => {
 
             describe('RouteUtils', () => {
                 test('RouteUtils.getGraphhopperBusParams', () => {
-                    const o = RouteUtils.getGraphhopperBusParams(end, start, time, arriveBy);
+                    const o = GhopperUtils.getGraphhopperBusParams(end, start, time, arriveBy);
```
Comment (young-k) [inline src/test/Unit.test.js]: not a fan of this variable naming, what is `o`

File: .gitignore (hunk 1 above)
Comment (zeladada) [reply .gitignore]: This is actually because it was ignoring the new GraphhopperUtils.js file

File: scripts/run_ghopper.sh (hunk 3 above)
Comment (zeladada) [reply scripts/run_ghopper.sh]: It's good to have in case you want filtered logs or console output

File: src/test/Unit.test.js (hunk 6 above)
Comment (zeladada) [reply src/test/Unit.test.js]: This is more readable, and it's just a test of a getter method, so naming is relatively unimportant.
image
image

**Pull request merged**